### ISSUE TITLE
makeflow: enable niceness via slurm

### ISSUE THREAD

**Issue by jsdillon** [cooperative-computing-lab/cctools#3795]
I'd like to be able to set the slurm niceness of all jobs makeflow submits to the queue when using makeflow. Something like:

```makeflow -T slurm -J 20 --nice 1000 my_makeflow.mf```

As far as I can tell, this isn't currently possible. Am I missing something?

**Comment by dthain** [comment]
Makeflow doesn't expose every thing the underlying batch system does, but I believe the`-B` option will add arbitrary things to the qsub command issued by make flow:
https://cctools.readthedocs.io/en/stable/man_pages/makeflow/#batch-options

So you could do something like this:
```
makeflow -T slurm -J 20 -B "--nice 1000" my_makeflow.mf
```

Does that solve your problem?

**Comment by jsdillon** [comment]
Thanks @dthain for the speedy reply! 

That really seems like it should have worked, but it does not seem to have. I tried both `makeflow -T slurm -J 20 -B "--nice 1000" my_makeflow.mf` and `makeflow -T slurm -J 20 -B "--nice=1000" my_makeflow.mf` and in both cases, the jobs were successfully submitted but with the niceness set to 0 (the default). 

However, I have verified that I can submit a script directly with `sbatch --nice=1000 my_script.sh` and it has the correct niceness (I believe the `=` is necessary in this syntax).

**Comment by dthain** [comment]
Hmm, I am actually pretty surprised that it did not work.  Could you try doing the same thing, but run makeflow with the option `-d all -o debug.log` to generate a debug log, and post that here?  That should make it clear exactly what command is being generated.

**Comment by dthain** [comment]
Yes, I suspect that is the problem: `-B` is an alias for `BATCH_OPTIONS` and I think the latter is overriding the former.

**Comment by jsdillon** [comment]
Got it. Thanks for your help. I think I know where to take it from here.

**Comment by dthain** [comment]
Glad to hear it -- If you don't mind, I'm going to convert this over to a discussion since. Think others may benefit from the answer.
